File: app/src/main/java/ru/michanic/mymot/Models/Manufacturer.java
package ru.michanic.mymot.Models;

import io.realm.RealmList;
import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;

public class Manufacturer extends RealmObject {

    @PrimaryKey
    private int id;

    private String code;
    private String image;
    private String name;
    private int sort;

    // связи
    private RealmList<Model> models;

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getImage() {
        return image;
    }

    public RealmList<Model> getModels() {
        return models;
    }
}

File: app/src/main/java/ru/michanic/mymot/Kotlin/Models/Manufacturer.kt
package ru.michanic.mymot.Kotlin.Models

import io.realm.RealmList
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import java.util.*

open class Manufacturer : RealmObject() {
    @PrimaryKey
    var id = 0
    var code: String? = null
    var image: String? = null
    var name: String? = null
    private var sort = 0

    var models: RealmList<Model>? = null
    val avitoSearchName: String
        get() = name!!.replace(" ", "%20").lowercase(Locale.getDefault())


    val autoruSearchName: String
        get() = "$code/"
}